consensus/safety-rules/src/safety_rules.rs

```diff
@@ -273,7 +334,11 @@ impl<T: Payload> TSafetyRules<T> for SafetyRules<T> {
                 .set_last_voted_round(timeout.round())?;
         }
 
-        let signature = timeout.sign(&self.validator_signer);
+        let validator_signer = self
```

Comment:
probably good to convert this into a function that returns a result
`fn signer(&self) -> Result<&ValidatorSigner, Error>`